perfkitbenchmarker/linux_benchmarks/spark_benchmark.py

```diff
@@ -74,23 +91,51 @@ def Run(benchmark_spec):
   """
   spark_cluster = benchmark_spec.spark_service
   jar_start = datetime.datetime.now()
-  success = spark_cluster.SubmitJob(FLAGS.spark_jarfile,
-                                    FLAGS.spark_classname)
-  if not success:
+
+  stdout_path = None
+  if FLAGS.spark_print_stdout:
+    # We need to get a name for a temporary file, so we create
+    # a file, then close it, and use that path name.
+    stdout_file = tempfile.NamedTemporaryFile(suffix='.stdout',
+                                              prefix='spark_benchmark',
+                                              delete=False)
+    stdout_path = stdout_file.name
+    stdout_file.close()
+    logging.info('job standard out will be written to ' + stdout_path)
+  stats = spark_cluster.SubmitJob(FLAGS.spark_jarfile,
+                                  FLAGS.spark_classname,
+                                  job_arguments=FLAGS.spark_job_arguments,
+                                  job_stdout_file=stdout_path)
+  if not stats[spark_service.SUCCESS]:
     raise Exception('Class {0} from jar {1} did not run'.format(
         FLAGS.spark_classname, FLAGS.spark_jarfile))
   jar_end = datetime.datetime.now()
-
-  metadata = spark_cluster.GetMetadata().update(
-      {'jarfile': FLAGS.spark_jarfile, 'class': FLAGS.spark_classname})
+  if stdout_path:
+    with open(stdout_path, 'r') as f:
+      logging.info('The output of the job is ' + f.read())
+    os.remove(stdout_path)
```

Comment:
Looks like there are cases where this file can be leaked - for instance if the job didn't complete successfully you raise an exception and skip executing this line.